In [1]:
# The following is from "big run"
import sys
sys.path.append('..')
from lib import *
import pandas as pd
import numpy as np

# Setup
lsa_np = np.load('../data/parsed/lsa_output.npy')

metadata = pd.read_pickle('../data/parsed/pickles/pickled_data_test.pickle')
metadata = metadata.loc[metadata['Scenario'] == '401']
metadata = metadata.reset_index(drop=True)

lsa_df = pd.DataFrame(lsa_np)

df = pd.concat([metadata, lsa_df], axis=1, join_axes=[metadata.index])
df = df.loc[df['Label'] != '-1']
df = df.reset_index(drop=True)

cat_features = ['To','From']
features = list(range(100))
features.extend(cat_features + ['Date'])

df = df[features + ['Label'] + ['ID']]


In [2]:
# optimal params
n_trees = 32
tree_depth = 70
random_seed = 42
n_max_features = 90
cat_features = ['To', 'From']

In [3]:
regular_increment(df, n_trees, tree_depth, random_seed, n_max_features, cat_features)

(0.5142857142857142, 0.9473684210526315, 0.82, 0.6666666666666666)
node has right child but no left
node has left child but no right
update finishes
(0.6, 0.8076923076923077, 0.81, 0.6885245901639345)
node has left child but no right
node has left child but no right
node has left child but no right
node has right child but no left
node has right child but no left
node has right child but no left
node has right child but no left
update finishes
(0.7714285714285715, 0.8181818181818182, 0.86, 0.7941176470588236)
update finishes
(0.8, 0.6666666666666666, 0.79, 0.7272727272727272)
update finishes
(0.7142857142857143, 0.6944444444444444, 0.79, 0.7042253521126761)
node has left child but no right
update finishes
(0.8, 0.6222222222222222, 0.76, 0.7000000000000001)


In [4]:
committee_increment(df, n_trees, tree_depth, random_seed, n_max_features, cat_features)

(0.5142857142857142, 0.9473684210526315, 0.82, 0.6666666666666666)
node has right child but no left
node has left child but no right
node has left child but no right
update finishes
(0.7428571428571429, 0.6842105263157895, 0.79, 0.7123287671232877)
node has left child but no right
update finishes
(0.8857142857142857, 0.775, 0.87, 0.8266666666666667)
node has right child but no left
node has left child but no right
update finishes
(0.6857142857142857, 0.75, 0.81, 0.7164179104477612)
update finishes
(0.5714285714285714, 0.6666666666666666, 0.75, 0.6153846153846153)
node has left child but no right
node has left child but no right
update finishes
(0.7428571428571429, 0.65, 0.77, 0.6933333333333334)


In [ ]:
homogeneous_increment(df, n_trees, tree_depth, random_seed, n_max_features, cat_features)

(0.5142857142857142, 0.9473684210526315, 0.82, 0.6666666666666666)
(0.9142857142857143, 0.6808510638297872, 0.82, 0.7804878048780487)
(0.8, 0.6511627906976745, 0.78, 0.7179487179487181)


In [3]:
most_confident_increment(df, n_trees, tree_depth, random_seed, n_max_features, cat_features)

(0.5142857142857142, 0.9473684210526315, 0.82, 0.6666666666666666)
['3.802951.M5SCFSRREEMMPFZ3C3UO2CLRXDYYFVIJB', '3.136531.DG5H5V3JPL4AKWOKUGF5TSJ1RNWQTL23A', '3.275105.MQQINY4X4KTF4ZL5HFYS1ZWCINV5UALDA', '3.554671.HQC1TCJG2ZJM1DWDXD5LXBPJLW4YJVMVA', '3.456452.CA4UXD4UTVT2NHJR3SABSEP30PKI2F2YA', '3.799909.PPROMXPLPLNU4IHAV3UYHTQKJF11ZHCVB', '3.802898.MQWUTBO0EKUW2KV05WCJRNFIAHGUUZ11B', '3.432720.DUAW3JD3E41N2STGWVKZJYRG5LUR4VGDA', '3.288157.LSBH0I0RO1KB11GN1AGXNZL1XYPMURFIA', '3.730010.BRH4OFED0AOVBOUR5SYZGVRMVTXHSXZ3B', '3.730722.IO2JZIAJHRV1YYMF5NPZJQDIQQGAQDBFB', '3.484357.MYUOGKW2YVEZNBYGDPTLE50IJYJQVQFTA', '3.138656.G4BS4WQLGSF2KMZB3CWY1EMCXPQNSFRRB', '3.881694.JVQZAMDZJFJESVN00CRTPF4Q01JZXMKPB', '3.905100.GEKLVYJ54E04RL4FUSMWOAESDUK5VJZPA', '3.19432.K5GLXFY143DGOAKIPLAVBA1UBKLBB5QFA', '3.829428.MWVN2MNDYSR2BTS0Z1FYH0HBJRDXEUQWB', '3.630813.GXSFJVW41XMKIWH5SUF2HEVZIG53P2LLB', '3.817329.KA1MXPN3UBAFBAYINYAQJ0M02CCZYKNFA', '3.284256.GLQIBL52FCZGO53VQJKJAV3AZXWY2QRRA', '3.684922.GPJ

ZeroDivisionError: division by zero

## Regular Increment Results  
(0.5142857142857142, 0.9473684210526315, 0.82, 0.6666666666666666)  
(0.6, 0.8076923076923077, 0.81, 0.6885245901639345)  
(0.7714285714285715, 0.8181818181818182, 0.86, 0.7941176470588236)  
(0.8, 0.6829268292682927, 0.8, 0.736842105263158)  
(0.7142857142857143, 0.6944444444444444, 0.79, 0.7042253521126761)  
(0.7428571428571429, 0.6666666666666666, 0.78, 0.7027027027027027)  

## Committee Increment Results  
(0.5142857142857142, 0.9473684210526315, 0.82, 0.6666666666666666)  
(0.7428571428571429, 0.6842105263157895, 0.79, 0.7123287671232877)  
(0.8, 0.6829268292682927, 0.8, 0.736842105263158)  
(0.5714285714285714, 0.7407407407407407, 0.78, 0.6451612903225806)  
(0.6857142857142857, 0.6666666666666666, 0.77, 0.676056338028169)  
(0.6857142857142857, 0.75, 0.81, 0.7164179104477612)  

## Homogeneous Increment Results  
(0.5142857142857142, 0.9473684210526315, 0.82, 0.6666666666666666)  
(0.9142857142857143, 0.6956521739130435, 0.83, 0.7901234567901234)  
(0.8, 0.6511627906976745, 0.78, 0.7179487179487181)  
(0.7714285714285715, 0.6585365853658537, 0.78, 0.7105263157894737)  
(0.7428571428571429, 0.7222222222222222, 0.81, 0.732394366197183)  
The last point is missing because the system ran out of rows to add

## Most Confident Increment Results  
